management/WFCORE-6830_Management_Resource_Constraints.adoc

```diff
@@ -0,0 +1,251 @@
+{
+    "outcome" => "success",
+    "result" => {
+        "allowed-origins" => undefined,
+        "backlog" => 50,
+        "connection-high-water" => 100,
+        "connection-low-water" => 75,
+        "console-enabled" => true,
+        "constant-headers" => undefined,
+        "http-authentication-factory" => "management-http-authentication",
+        "http-upgrade" => {
+            "enabled" => true,
+            "sasl-authentication-factory" => "management-sasl-authentication"
+        },
+        "http-upgrade-enabled" => true,
+        "no-request-timeout" => 60000,
+        "sasl-protocol" => "remote",
+        "secure-socket-binding" => undefined,
+        "server-name" => undefined,
+        "socket-binding" => "management-http",
+        "ssl-context" => undefined
+    }
+}
+----
+
+When persisted to the configuration file the resulting XML will look like:
+
+[source,xml]
+----
+<management-interfaces>
+    <http-interface http-authentication-factory="management-http-authentication"
+        backlog="51" no-request-timeout="60001"
+        connection-high-water="101" connection-low-water="76">
+
+        <http-upgrade enabled="true" sasl-authentication-factory="management-sasl-authentication"/>
+        <socket-binding http="management-http"/>
+    </http-interface>
+</management-interfaces>
+----
+
+=== Nice-to-Have Requirements
+
+None
+
+=== Non-Requirements
+
+The purpose of this feature request is to expose management model attributes to configure existing behaviour within the
+application server, this feature request is not intended to revisit the existing behaviour which is already available at
+all stability levels.
+
+=== Future Work
+
+The only possible future work could be to promote one or all of these new attributes to a higher stability level.
+
+A change such as this should likely be constrained to:
+
+ * Removing the stability level qualifier from the attribute definitions allowing them to be used at the default stability level.
+ * Moving the XML schema definitions and parsing to an appropriate schema at the default stability level.
+ * The community documentation can be simplified to remove references to the stability level
+ `Community` once supported at all stability levels.
+
+== Backwards Compatibility
+
+In considering backwards compatibility if an attribute is defined on the HTTP management interface AND the
+corresponding system property is set then the system property should take precedence.
+
+For each of the system properties if defined and if the attribute is supported at the current stability level  a message in the
+following format will be logged to inform the administrator that they should move to using the new attribute definitions.
+
+[source]
+----
+14:58:13,878 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0515: The system
+property 'org.wildfly.management.backlog' is deprecated and may be removed in a future version, attribute 'backlog' on
```

Comment:
s/version, attribute/version. The attribute/g

This is more a comment for the impl PR, so don't feel the need to change this if you wouldn't otherwise.